### ISSUE TITLE
ELR'd squad loses ability to battle harden

### ISSUE THREAD

**Issue by jrvjrv** [vasl-developers/vasl#331]
Drag a Soviet 4-5-8 onto the map. Press ctrl-E. The squad ELRs to a 4-4-7. Right-click; in the HoB submenu "battle harden" is available. Press ctrl-E again. The squad is replaced with a 4-2-6. Right-click; "battle harden" is no longer available in the HoB submenu (nor does ctrl-Q work).

A 4-2-6 dragged directly from the palette has "battle harden" available, & ctrl-Q works.

vasl version 6.3.3

**Comment by jrvjrv** [comment]
This may be a deliberate "feature" or workaround. As I understand the code, the Soviet 4-5-8 ELR progression is defined through layers. The 4-5-8 ELRs to a 4-4-7 ELRs to a 4-2-6. But once it reaches 4-2-6 it should battle harden to a 5-2-7 instead of a 4-4-7. This creates two types of 4-2-6: ones that are on the 4-5-8 track and ones that are on the 6-2-8 track. The 4-5-8 track 4-2-6 can't go back up the ladder. 

In short I think I understand why it works the way it does. It's still a defect. It ought to work the "right" way, and as a vasl user I would be mightily mystified by the behavior without the above understanding.

**Comment by zgrose** [comment]
I'd have to dig a little closer, but I believe you have discovered the flaw in the layer approach to unit replacement that gave us fits for PiF.

**Comment by derimmer** [comment]
Closing this one as it is duplicated by #862 with exactly the same diagnosis.
